jni/src/faiss_wrapper.cpp

```diff
@@ -409,6 +412,22 @@ jobjectArray knn_jni::faiss_wrapper::QueryIndex_WithFilter(knn_jni::JNIUtilInter
     return results;
 }
 
+int getQueryEfSearch(JNIEnv * env, knn_jni::JNIUtilInterface * jniUtil, jobject algoParams, int defaultEfSearch) {
+    if (algoParams == nullptr) {
+        return defaultEfSearch;
+    }
+    std::string hnswParameterJavaFilePath = "org/opensearch/knn/index/query/model/HNSWAlgoQueryParameters";
+    if (jniUtil->IsInstanceOf(env, algoParams, hnswParameterJavaFilePath)) {
```

Comment:
Is there any way to avoid multiple java calls?

Reply:
Not that I know of. We will have to go back to using primitive types on JNI layer by unpacking it in the java layer. How much is the overhead comparatively considering we are caching classes on initialization?

Reply:
> How much is the overhead comparatively considering we are caching classes on initialization?

From performance, Im not sure. But we are
1. Checking if instance type
2. Getting Optional

Why not let java pass a map to the JNI with the parameters?

Reply:
If the concern is not performance, whats the benefit of the map when we can have the entire object? Whats the concern with using an object

The map becomes a string to string property bag which will force us to validate again on the JNI layer and there is always a possibility of unknown parameters being added on accident. A typed value is seems  a better approach than passing around a generic map